Documentation/BasicOperators.md

```diff
@@ -46,41 +46,416 @@ code-style will be used.
 ## Performing side effects with signals
 
 ### Observation
+
+`Signal`s can be observed with the `observe` function. It takes an `Observer` as argument to which any future events are sent. 
+
+```Swift
+signal.observe(Signal.Observer { event in
+    switch event {
+    case let .Next(next):
+        println("Next: \(next)")
+    case let .Error(error):
+        println("Error: \(error)")
+    case .Completed:
+        println("Completed")
+    case .Interrupted:
+        println("Interrupted")
+    }
+})
+```
+
+Alternatively, callbacks for the `Next`, `Error`, `Completed` and `Interrupted` events can be provided which will be called when a corresponding event occurs.
+
+```Swift
+signal.observe(next: { next in
+        println("Next: \(next)")
+    }, error: { error in
+        println("Error: \(error)")
+    }, completed: {
+        println("Completed")
+    }, interrupted: {
+        println("Interrupted")
+    }
+)
+```
+
+Note that it is not necessary to provide all four parameters - all of them are optional, you only need to provide callbacks for the events you care about.
+
+`observe` is also available as operator that can be used with [|>](#pipe)
+
 ### Injecting effects
 
+Side effects can be injected on a `SignalProducer` with the `on` operator without actually subscribing to it. 
+
+```Swift
+let producer = signalProducer
+    |> on(started: {
+	        println("Started")
+        }, event: { event in
+            println("Event: \(event)")
+        }, error: { error in
+            println("Error: \(error)")
+        }, completed: {
+            println("Completed")
+        }, interrupted: {
+            println("Interrupted")
+        }, terminated: {
+            println("Terminated")
+        }, disposed: {
+            println("Disposed")
+        }, next: { next in
+            println("Next: \(next)")
+    })
```

Comment:
Is it correct at the same indentation level as the `|>` or should it be completely to the left?